### DISCUSSION TITLE
Why is the default implementation for train_dataloader in DataHooks a warning?

### DISCUSSION THREAD

**Discussion by ananthsub** [Lightning-AI/pytorch-lightning#8734] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
The default implementation for these is logging a warning that nothing is implemented. https://github.com/PyTorchLightning/pytorch-lightning/blob/963c26764682fa4cf64c93c5a7572ae0040e9c32/pytorch_lightning/core/hooks.py#L529

Why isn’t the default implementation to raise a NotImplementedError? This would make errors much clearer in case users forget to override these hooks

**Comment by carmocca** [comment, marked as answer]
I believe this is just legacy code. No real reason. It's in the original implementation (to bolts!)

https://github.com/PyTorchLightning/lightning-bolts/commit/797464c421652c4ceb4e8527b9f56ab059382200

Feel free to try changing it :)

**Reply by ananthsub** [reply]
filed https://github.com/PyTorchLightning/pytorch-lightning/issues/8752 as a followup
